zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/StreamProcessor.java

```diff
@@ -464,19 +466,24 @@ public ActorFuture<Long> getLastWrittenPositionAsync() {
   @Override
   public HealthReport getHealthReport() {
     if (actor.isClosed()) {
-      return HealthReport.unhealthy(this).withMessage("actor is closed");
+      return HealthReport.unhealthy(this)
+          .withMessage("actor is closed", streamProcessorContext.getClock().instant());
     }
 
     if (processingStateMachine != null && !processingStateMachine.isMakingProgress()) {
       return HealthReport.unhealthy(this)
-          .withMessage("Processing not making progress. It is in an error handling loop.");
+          .withMessage(
+              "Processing not making progress. It is in an error handling loop.",
+              streamProcessorContext.getClock().instant());
```

Comment:
:thinking: Interesting. The `StreamClock` is controllable and used mostly for externally visible things - e.g. writing when a command was processed on the log, for example, or for scheduling of BPMN timers. This allows users to "time travel" in their processes, which is very useful for testing (e.g. "trigger this timer on November 24th 2031", and the corresponding events and so on are at the time). It's divorced from the actor clock or system clock to avoid triggering timeouts and other time-sensitive things in other parts, where we need wall time/logical time (e.g. Raft leader election).

In this case, we would likely want to show the actor clock, I think? This is more about internal failures than user processing, so I would think that's the right clock to use. `ActorClock.instant()`.